Increase test coverage up to 100% - phase 1 :)

simahawk opened the pull request.

simahawk committed:
cms_notification: test coverage 100%

simahawk committed:
cms_notification: fix default help templates

While migrating to v11 the help templates for notification options
had not been updated properly but we did not notice it
because the widget is fault tolerant.

Fixed and added specific test for them.

simahawk committed:
cms_info: test coverage 100%

simahawk committed:
cms_account_form: test coverage 100%

simahawk committed:
cms_info: info dict default to None values

simahawk committed:
cms_toolbar: test coverage 100%

simahawk committed:
cms_form: split widget models to sep files

simahawk committed:
cms_form: widgets test cov 100%

simahawk committed:
cms_form: fix binary widget ignore none values

simahawk committed:
gitignore: ignores `.eggs` folders

simahawk committed:
cms_form: x2many widget test cov + fix value handling

simahawk committed:
cms_form: binary widget test cov + fix value handling

simahawk committed:
cms_form: utils test cov 100%

simahawk committed:
cms_form: cms.form.mixin test cov 100%

leemannd reviewed (commented):
Just did a small check and I have a question:
- not sure it's OCA proof `# pragma: no cover`
- I saw a few `TODO` do you intend to address them in this PR, or another one?

simahawk commented:
`pragma: no cover` is perfectly fine when you don't care about testing
`TODO` will be handled when they will... no specific relation to this PR

simahawk committed:
cms_form: fix selection widget w/ non-Selection field

Potentially you can apply form widgets to different type of fields.
Easy case: apply a select to an integer field.
In such case we should not rely on `selection` key to be there
when extracting the value.

simahawk commented:
merging this as it's keeping blocked other PRs and new improvements/fixes/test cov.
I'm going to get back to this subject in phase 2.
